presto-geospatial/src/main/java/com/facebook/presto/plugin/geospatial/SphericalGeometryType.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.plugin.geospatial;
+
+import com.facebook.presto.spi.ConnectorSession;
+import com.facebook.presto.spi.block.Block;
+import com.facebook.presto.spi.block.BlockBuilder;
+import com.facebook.presto.spi.type.AbstractVariableWidthType;
+import com.facebook.presto.spi.type.TypeSignature;
+import io.airlift.slice.Slice;
+
+import static com.facebook.presto.geospatial.serde.GeometrySerde.deserialize;
+
+/**
+ * Class representing a SphericalGeometry value in SQL. It supports the same set of geometry shapes as GeometryType, e.g.
+ * Point, Multipoint, Linestring, Multilinestring, Polygon, Multipolygon, etc., but assumes they're shapes on the spherical
+ * surface of earth, following the WGS84 reference system (http://www.unoosa.org/pdf/icg/2012/template/WGS_84.pdf). Geo
+ * functions applied to SphericalGeometry objects operate in the spherical space and return values in meters (e.g. for distance)
+ * and square meters (e.g. for area).
+ */
+public class SphericalGeometryType
+        extends AbstractVariableWidthType
+{
+    public static final SphericalGeometryType SPHERICAL_GEOMETRY = new SphericalGeometryType();
+    public static final String SPHERICAL_GEOMETRY_TYPE_NAME = "SphericalGeometry";
+
+    public static final float MIN_LAT = -90;
```

Comment:
s/MIN_LAT/MIN_LATITUDE, ditto other variables

These values are used only in `GeoFunctions.java`. Move them there.